### DISCUSSION TITLE
Has anyone successfully used React Native?

### DISCUSSION THREAD

**Discussion by bbhxwl** [ultralytics/ultralytics#20941] [Q&A]
Has anyone successfully used React Native?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @bbhxwl, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response. An Ultralytics engineer will also assist you soon 😊.

**Reply by glenn-jocher** [reply]
Hello @bbhxwl! While we don't have specific React Native integration examples in our documentation, YOLO11 models can definitely be used in mobile apps built with React Native. You'd typically export your model to mobile-friendly formats like [ONNX](https://docs.ultralytics.com/integrations/onnx/), CoreML (iOS), or TensorFlow Lite using our [export functionality](https://docs.ultralytics.com/modes/export/), then integrate them using React Native bridges or libraries like ONNX Runtime React Native. Alternatively, you could use our cloud inference APIs for easier integration. I'd encourage other community members who have React Native experience to share their approaches here!

**Reply by bbhxwl** [reply]
> Hello @bbhxwl! While we don't have specific React Native integration examples in our documentation, YOLO11 models can definitely be used in mobile apps built with React Native. You'd typically export your model to mobile-friendly formats like [ONNX](https://docs.ultralytics.com/integrations/onnx/), CoreML (iOS), or TensorFlow Lite using our [export functionality](https://docs.ultralytics.com/modes/export/), then integrate them using React Native bridges or libraries like ONNX Runtime React Native. Alternatively, you could use our cloud inference APIs for easier integration. I'd encourage other community members who have React Native experience to share their approaches here!

Do you have any code examples for onnx? Not React native is also acceptable. Python is also possible, I can refer to writing in other languages

**Reply by glenn-jocher** [reply]
@bbhxwl not for React Native, but we have Flutter for mobile here: https://github.com/ultralytics/yolo-flutter-app

**Reply by bbhxwl** [reply]
> @bbhxwl not for React Native, but we have Flutter for mobile here: https://github.com/ultralytics/yolo-flutter-app

Are there any other language demonstrations available? I can't understand Flutter code

**Reply by bbhxwl** [reply]
> @bbhxwl not for React Native, but we have Flutter for mobile here: https://github.com/ultralytics/yolo-flutter-app

I can use Node.js, Python, and C#

**Reply by glenn-jocher** [reply]
@bbhxwl here's a Python example for ONNX export and inference:

```python
from ultralytics import YOLO

# Load the YOLO11 model
model = YOLO("yolo11n.pt")

# Export the model to ONNX format
model.export(format="onnx")  # creates 'yolo11n.onnx'

# Load the exported ONNX model
onnx_model = YOLO("yolo11n.onnx")

# Run inference
results = onnx_model("path/to/image.jpg")

# Process results
for r in results:
    print(r.boxes)  # print bounding box predictions
```

For Node.js and C#, you'd use the ONNX Runtime libraries specific to those languages - the exported `.onnx` file works across all platforms once you have the appropriate runtime installed.
